src/NoteDataUtil.cpp

```diff
@@ -1855,6 +1855,144 @@ static void GetTrackMapping( StepsType st, NoteDataUtil::TrackMapping tt, int Nu
 	}
 }
 
+static void HyperShuffleNotes( NoteData &inout, int iStartIndex, int iEndIndex)
+{
+	int iNumTracks = inout.GetNumTracks();
+
+	// Stores the list of tracks that can have notes placed into them.
+	// This will be shuffled so that notes can be placed in different tracks.
+	vector<int> viTargetTracks;
+	// Stores the list of tap notes that need to be placed.
+	vector<TapNote> vtnTargetTaps;
+	viTargetTracks.reserve(iNumTracks);
+	vtnTargetTaps.reserve(iNumTracks);
+
+	vector<int> viHoldEndRows;
```

Comment:
These three lines can be combined and done in one line with `std::vector<int> viHoldEndRows(iNumTracks, -1)'`

Reply:
The first two are different vectors so I'm not sure how i could combine those (unless you are saying do them the same way.

Reply:
I meant specifically for the things for `viHoldEndRows` (lines 1870-1872)

Reply:
(my suggestion creates a vector with `iNumTracks` elements and fills it with `-1`)